packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/ClientEventService.java

```diff
@@ -156,51 +156,56 @@ private void send(final Object message, final List<SseEmitter> emitterList, fina
 				log.error("Error sending all users message to user {}", userId, e);
 			}
 		});
-		emittersToRemove.forEach((emitter) -> {
-			userIdToEmitter.get(userId).remove(emitter);
-			if (userIdToEmitter.get(userId).size() == 0) {
-				userIdToEmitter.remove(userId);
-			}
-		});
+
+		synchronized (userIdToEmitter) {
+			emittersToRemove.forEach((emitter) -> {
+				userIdToEmitter.get(userId).remove(emitter);
+				if (userIdToEmitter.get(userId).size() == 0) {
+					userIdToEmitter.remove(userId);
+				}
+			});
+		}
 	}
 
 	/**
-	 * Lisens for messages to send to a user and if we have the SSE connection, send it
+	 * Lisens for messages to send to a user and if we have the SSE connection, send
+	 * it
 	 *
 	 * @param message the message to send
 	 * @param channel the channel to send the message on
 	 * @throws IOException if there was an error sending the message
 	 */
 	// TODO: use anonymous queues, currently this wont behave correctly with
 	// multiple hmi-server instances. Issue #2679
-	@RabbitListener(
-			queues = {"${terarium.client-user-event-queue}"},
-			concurrency = "1")
+	@RabbitListener(queues = { "${terarium.client-user-event-queue}" }, concurrency = "1")
 	void onSendToUserEvent(final Message message, final Channel channel) throws IOException {
 		final JsonNode messageJson = decodeMessage(message, JsonNode.class);
 		if (messageJson == null) {
 			return;
 		}
+
 		final String userId = messageJson.at("/userId").asText();
-		final List<SseEmitter> emitterList = userIdToEmitter.get(userId);
 		synchronized (userIdToEmitter) {
-			send(messageJson.at("/event"), emitterList, userId);
+			final List<SseEmitter> emitterList = userIdToEmitter.get(userId);
+			if (emitterList != null) {
```

Comment:
If the `userId` was missing, or if there were no emitters for the user id, that list would result in a NPE.

Reply:
Ah... that is a smart catch.  Also explains why we probably don't see it locally as it is a curious race condition